qrutyy/OcamlADT
# Implement draft of qcheck testing + pretty-printer

#### Implement enhanced pretty printer for improved formatting. Add tests for functionality validation. Fix existing issues for cleaner output. Incorporate QCheck for generating abstract syntax trees (AST). Update build system to support new features. Improve code readability and reliability through testing.

TODO: 
- fix generator (limit the size)
- test on new parser
- add more cases
- add success print for manual check
